### DISCUSSION TITLE
Use high-level guards with object guards

### DISCUSSION THREAD

**Discussion by john891226** [statelyai/xstate#4893] [General]
Hi, I'm struggling to use the: "not" high-level guard with a parametrized guard, but can't manage to use it with an object guard for defining the params method. Is there any other way of archiving this currently?

Current xstate version: 5.12.0

![image](https://github.com/statelyai/xstate/assets/15107504/6d867a30-94e7-44e8-9f35-1c757425fbbb)

**Comment by davidkpiano** [comment]
cc. @Andarist

**Comment by dstn3422** [comment]
Hi.

Since xState 6 will only offer this syntax will it be impossible to use named guards with params inside higher order guards without a typescript error?
